tools/build_pak_files.py

```diff
@@ -57,21 +51,44 @@
-        print(f"[SPEAR | build_pak_files.py] File or directory or symlink exists, removing: {unreal_project_content_shared_dir}")
-        spear.remove_path(unreal_project_content_shared_dir)
+    # We use different strategies for setting scene_ids, depending on if we're creating symlinks or not.
+    # If we're not creating symlinks, then the user must specify args.scene_ids. If we are creating
+    # symlinks, then we get a list of candidate scene_ids from Perforce and optionally filter.
+    if args.skip_create_symlinks:
+        assert args.scene_ids is not None
+        scene_ids = [args.scene_ids]
+    else:
+        assert os.path.exists(args.perforce_content_dir)
+        perforce_content_scenes_dir = os.path.realpath(os.path.join(args.perforce_content_dir, "Scenes"))
+        assert os.path.exists(perforce_content_scenes_dir)
+
+        ignore_names = [".DS_STORE"]
+        scene_ids = [ os.path.basename(x) for x in os.listdir(perforce_content_scenes_dir) if x not in ignore_names ]
+        assert len(scene_ids) > 0
+        if args.scene_ids is not None:
+            scene_ids = [ s for s in scene_ids if fnmatch.fnmatch(s, args.scene_ids) ]
+        assert len(scene_ids) > 0
+
+    # Create a symlink to the Shared directory
+    if not args.skip_create_symlinks:
+        perforce_content_shared_dir = os.path.realpath(os.path.join(args.perforce_content_dir, "Shared"))
+        assert os.path.exists(perforce_content_shared_dir)
 
-    print(f"[SPEAR | build_pak_files.py] Creating symlink: {unreal_project_content_shared_dir} -> {perforce_content_shared_dir}")
-    os.symlink(perforce_content_shared_dir, unreal_project_content_shared_dir)
+        # We do not want to use os.path.realpath(...) here, because that will resolve to the directory inside the user's Perforce workspace.
+        # Instead, we want this path to refer to the symlinked version inside the user's unreal project directory.
+        unreal_project_content_shared_dir = os.path.join(unreal_project_content_dir, "Shared")
 
-    assert os.path.exists(perforce_content_scenes_dir)
-    scene_ids = [ os.path.basename(x) for x in os.listdir(perforce_content_scenes_dir) ]
-    assert len(scene_ids) > 0
+        if spear.path_exists(unreal_project_content_shared_dir):
+            print(f"[SPEAR | build_pak_files.py] File or directory or symlink exists, removing: {unreal_project_content_shared_dir}")
+            spear.remove_path(unreal_project_content_shared_dir)
 
-    if args.scene_ids is not None:
-        scene_ids = [ s for s in scene_ids if fnmatch.fnmatch(s, args.scene_ids) ]
+        print(f"[SPEAR | build_pak_files.py] Creating symlink: {unreal_project_content_shared_dir} -> {perforce_content_shared_dir}")
+        os.symlink(perforce_content_shared_dir, unreal_project_content_shared_dir)
 
-    assert len(scene_ids) > 0
+    # We do not want to use os.path.realpath(...) here, because that will resolve to the directory inside the user's Perforce workspace.
```

Comment:
Move this up so it's logically next to `unreal_project_cooked_dir`.

```
    # Once we know the platform, we can set our cooked dir.
    unreal_project_cooked_dir = os.path.realpath(os.path.join(unreal_project_dir, "Saved", "Cooked", platform + "NoEditor"))

    # We do not want to use os.path.realpath(...) here, because that will resolve to the directory inside the user's Perforce workspace.
    # Instead, we want this path to refer to the symlinked version inside the user's unreal project directory.
    unreal_project_content_scenes_dir = os.path.join(unreal_project_content_dir, "Scenes")
    assert os.path.exists(unreal_project_content_scenes_dir)
```